fuzzamoto/src/targets/bitcoin_core.rs

```diff
@@ -226,13 +229,116 @@ impl ConnectableTarget for BitcoinCoreTarget {
     }
 }
 
-impl HasTipHash for BitcoinCoreTarget {
-    fn get_tip_hash(&self) -> Option<[u8; 32]> {
+impl HasGetBestBlockHash for BitcoinCoreTarget {
+    fn getbestblockhash(&self) -> Option<BlockHash> {
         match self.node.client.get_best_block_hash() {
-            Ok(result) => result
-                .block_hash()
-                .ok()
-                .map(|h| *h.as_raw_hash().as_byte_array()),
+            Ok(result) => result.block_hash().ok(),
+            Err(_) => None,
+        }
+    }
+}
+
+impl HasGetBlockCount for BitcoinCoreTarget {
+    fn getblockcount(&self) -> Option<u64> {
+        match self.node.client.get_block_count() {
+            Ok(result) => Some(result.0),
+            Err(_) => None,
+        }
+    }
+}
+
+impl HasGetBlock for BitcoinCoreTarget {
+    fn getblock(&self, hash: BlockHash) -> Option<Block> {
+        match self.node.client.get_block(hash) {
+            Ok(result) => Some(result),
+            Err(_) => None,
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Mempool {
```

Comment:
```suggestion
/// Information about a transaction in the mempool
#[derive(Clone, Debug)]
pub struct MempoolEntry {
```

"Mempool" refers to the whole data structure but this type just refers to one entry and some info about it